src/rt-app.c

```diff
@@ -838,6 +842,7 @@ void *thread_body(void *arg)
 			data->ind, policy_to_string(data->sched_data->policy),
 			data->sched_data->prio);
 	set_thread_priority(data, data->sched_data);
+	set_taskgroup(data, data->taskgroup);
```

Comment:
My impression was that this should work OK.

Reply:
IMHO, it's different with set_thread_priority(). In case the task (or the default block) specifies sched_data (policy, priority, etc.) and the first phase of that task does not, then you want this task to be run with the sched_data of the task (or the default block) during the first phase already.

```
{
 	"tasks" : {
 		"task0" : {
 			"loop" : 2,
			"policy" : "SCHED_FIFO",
			"phases" : {
 				"p0" : {
					"run" : 2000,
					"sleep" : 8000
 				},
 				"p1" : {
					"run" : 2000,
					"sleep" : 8000,
					"policy" : "SCHED_OTHER"
 				}
			}
		}
	},
	"global": {
 		"calibration": "CPU0"
 	}
}
```

```[rt-app] <notice> [0] Starting with SCHED_FIFO policy with priority 10
[rt-app] <debug> [0] setting scheduler SCHED_FIFO priority 10
[rt-app] <debug> [0] begins thread_loop 0 phase 0 phase_loop 0
[rt-app] <debug> [0] runs events 0 type 6 
[rt-app] <debug> run 2000 
[rt-app] <debug> [0] runs events 1 type 5 
[rt-app] <debug> sleep 8000 
[rt-app] <debug> [0] setting scheduler SCHED_OTHER priority 0
[rt-app] <debug> [0] begins thread_loop 0 phase 1 phase_loop 0
[rt-app] <debug> [0] runs events 0 type 6 
[rt-app] <debug> run 2000 
[rt-app] <debug> [0] runs events 1 type 5 
[rt-app] <debug> sleep 8000 
[rt-app] <debug> [0] begins thread_loop 1 phase 0 phase_loop 0
[rt-app] <debug> [0] runs events 0 type 6 
[rt-app] <debug> run 2000 
[rt-app] <debug> [0] runs events 1 type 5 
[rt-app] <debug> sleep 8000 
[rt-app] <debug> [0] begins thread_loop 1 phase 1 phase_loop 0
[rt-app] <debug> [0] runs events 0 type 6 
[rt-app] <debug> run 2000 
[rt-app] <debug> [0] runs events 1 type 5 
[rt-app] <debug> sleep 8000 
[rt-app] <notice> [0] Exiting.
```
You can see that with phase 1 runs in SCHED_FIFO in the first loop and in SCHED_OTHER in the second loop.